store/store.go

```diff
@@ -680,6 +694,33 @@ func (c *Client) GetLatestPumpSettingsV1(ctx context.Context, traceID string, us
 	return dataCollection(c).Find(ctx, query, opts)
 }
 
+func (c *Client) GetLatestBasalSecurityProfile(ctx context.Context, traceID string, userID string) (*DbProfile, error) {
+	if userID == "" {
+		return nil, errors.New("invalid user id")
+	}
+
+	query := bson.M{
+		"_userId": userID,
+		"type":    "basalSecurity",
+	}
+	opts := options.FindOne()
+	//opts.SetProjection(unwantedPumpSettingsFields) TODO
+	opts.SetSort(bson.M{"time": -1})
+	opts.SetComment(traceID)
+	var result *DbProfile
+	err := dataCollection(c).FindOne(ctx, query, opts).Decode(&result)
```

Comment:
Testing the API, I get cannot decode nil values errors
```suggestion
	var result DbProfile
	err := dataCollection(c).FindOne(ctx, query, opts).Decode(&result)
```
and `return &result, nil` at the end